src/char/int_mail.cpp

```diff
@@ -361,9 +361,6 @@ void mapif_Mail_getattach(int fd, uint32 char_id, int mail_id, int type)
 	if( msg.dest_id != char_id )
 		return;
 
-	if( msg.status != MAIL_READ )
```

Comment:
This should be PACKETVER dependent!
Although I do not know when this new button was added. :-(
But before that button a player would not have been able to get attachments from a mail without opening it, that's basically why that check existed.
Another option would be offering a config to let admins choose if they allow it or not?